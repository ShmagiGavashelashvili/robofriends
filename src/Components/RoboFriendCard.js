import React from "react";

function RoboFriendCard({ id, username, name, email, random }) {
  return (
    <div className="robofriends_card">
      <img src={`https://robohash.org/${random}${id}?size=200x200`} alt={username} />
      <h2>{name}</h2>
      <p data-testid="email">{email}</p>
    </div>
  );
}

export default RoboFriendCard;
