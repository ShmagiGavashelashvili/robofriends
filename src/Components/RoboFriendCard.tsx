import React from "react";

interface Props {
  id: number,
  username: string,
  name: string,
  email: string,
  random: number | null
}

function RoboFriendCard({ id, username, name, email, random }: Props) {
  return (
    <div className="robofriends_card">
      <img src={`https://robohash.org/${random}${id}?size=200x200`} alt={username} />
      <h2>{name}</h2>
      <p data-testid="email">{email}</p>
    </div>
  );
}

export default RoboFriendCard;
