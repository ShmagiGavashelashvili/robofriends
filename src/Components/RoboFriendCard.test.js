import React from "react";
import { render } from "@testing-library/react";

import RoboFriendCard from "./RoboFriendCard";

describe("RoboFriendCard", () => {
  // let robot = {
  //   name: "shmagi",
  //   id: "1",
  //   username: "shmagi robo",
  //   email: "shmagi@gmail.com",
  //   random: 6,
  // };
  test("test component render", () => {
    render(<RoboFriendCard />);

    // const name = screen.getByRole("heading");
    // const email = screen.getByTestId("email");
    // const img = screen.getByRole("img");

    // expect(name).toBeInTheDocument();
    // expect(email).toBeInTheDocument();
    // expect(img).toBeInTheDocument();
    // expect(name).toHaveTextContent("shmagi");
    // expect(email).toHaveTextContent("shmagi@gmail.com");
    expect(render(<RoboFriendCard />)).toMatchSnapshot();
  });
});
