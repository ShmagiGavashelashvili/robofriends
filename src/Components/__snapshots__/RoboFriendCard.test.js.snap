// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoboFriendCard test component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="robofriends_card"
      >
        <img
          src="https://robohash.org/undefinedundefined?size=200x200"
        />
        <h2 />
        <p
          data-testid="email"
        />
      </div>
    </div>
    <div>
      <div
        class="robofriends_card"
      >
        <img
          src="https://robohash.org/undefinedundefined?size=200x200"
        />
        <h2 />
        <p
          data-testid="email"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="robofriends_card"
    >
      <img
        src="https://robohash.org/undefinedundefined?size=200x200"
      />
      <h2 />
      <p
        data-testid="email"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
